extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationServiceSpec.js

```diff
@@ -191,4 +225,42 @@ describe('ItemSelectionInputValidationService', function() {
           'rule 1, the "proper subset" rule must include at least 2 options.')
       }]);
     });
+
+  it(
+    'should expect min possible selections to be less than or equal to ' +
+    'number of correct options when "Equals" rule is used.',
+    function() {
+      // Make min allowed selections greater than correct answers
+      customizationArguments.minAllowableSelectionCount.value = 2;
+
+      var warnings = validatorService.getAllWarnings(
+        currentState, customizationArguments, OneInputAnswerGroups,
+        goodDefaultOutcome);
+      expect(warnings).toEqual([{
+        type: WARNING_TYPES.ERROR,
+        message: (
+          'In answer group 1, ' +
+          'rule 1, the "Equals" rule should not have more correct options ' +
+          'than minimum allowed selection count.')
+      }]);
+    });
+
+  it(
+    'should expect at lease one option when  ' +
+    '"ContainsAtLeastOneOf" rule is used.',
```

Comment:
The customisation args work for all the rules. So, instead of just checking that it should have at least 1 option take care that selected options should be >= minAllowedCount and <= maxAllowedCount. Also, minAllowedCount should be >=1 in all cases.

Reply:
Hi @vibhor98 
Even if the selected options are < min allowed count, the user can still choose some option that is in the given answer set (if the answer set has at least 1 value, which is what I am checking). And the same applies even if it is greater than maxAllowedCount.

Reply:
Yes, agreed! Since the name of the rule is itself- 'ContainsAtLeastOneOf' we can exclude this rule from checking the lower and upper bound for the selected choices.